### PR TITLE
Jcollard/add vagrantfile

### PR DESCRIPTION
**Pull request by jcollard**
I created a Vagrant development environment for this project.

It installs the necessary dependencies in a VM, sets up file synch, and port forwarding.

### CODE REVIEW

File: vagrant/Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+# you're doing.
```
Comment (treznick) [inline vagrant/Vagrantfile]: what does this comment do?

File: vagrant/base/Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    apt-get install -y ruby ruby-dev cmake build-essential zlib1g-dev
+    while [ $? -ne 0 ]; do
+       sleep 1
+       apt-get install -y ruby ruby-dev cmake build-essential zlib1g-dev
+    done
+    gem install bundler
+    source /home/vagrant/.nvm/nvm.sh; npm -g install yarn
+
+  SHELL
+
```
Comment (treznick) [inline vagrant/base/Vagrantfile]: unnecessary whitespace

File: vagrant/README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # Vagrant Development Environment
 
-This development environment is designed to allow you to start working quickly and from any OS which can install (Vagrant)[https://www.vagrantup.com/] and (VirtualBox)[https://www.virtualbox.org/wiki/Downloads].
+This development environment is designed to allow you to start working quickly and from any OS which can install [Vagrant](https://www.vagrantup.com/) and [VirtualBox](https://www.virtualbox.org/wiki/Downloads).
```
Comment (treznick) [inline vagrant/README.md]: I'd fix this up into the prior commit, no?

File: vagrant/README.md [hunk 4]
```diff
@@ -6,7 +6,7 @@ This box comes packaged with all of the dependencies necessary to build and run
 
 ## Quick Start
 
-1. Ensure you have Vagrant and VirtualBox installed
+1. Ensure you have [Vagrant](https://www.vagrantup.com/) and [VirtualBox](https://www.virtualbox.org/wiki/Downloads) installed
```
Comment (treznick) [inline vagrant/README.md]: I'd also fix this up into the prior commit

**Comment by jcollard** [comment]
I'd recommend giving it a try at some point. It is a widely used tool that provides utility for sand boxing development environments. I have found it particularly useful when working on projects for many different clients and when working in multi-person projects.

Benefits of using Vagrant / providing a Vagrant environment for collaborators:
* Easy to have your development environment mimic your production environment

* Isolated development environment won't interfere with other projects (e.g. multiple versions of python / ruby / java / apache / etc).

* New developers can start working on the project immediately, on any OS, without the need to install specific gems, a specific version of node, a specific version of npm, specific libraries dependent on the project dependencies. For example, on this project, the following libraries are required to be installed `ruby ruby-dev cmake build-essential zlib1g-dev` to be able to build the project.

* Makes it easier to provide support when someone raises an Issue by being able to replicate the exact environment of the user.

File: vagrant/Vagrantfile (hunk 1 above)
Comment (jcollard) [reply vagrant/Vagrantfile]: The first 2 line are generated when creating a Vagrant file using `vagrant up`. These lines tell different editing tools that they should use ruby highlighting if it is available.

The third line is a missed deletion of another comment.

File: vagrant/README.md (hunk 3 above)
Comment (jcollard) [reply vagrant/README.md]: Sure. Unfortunately, I didn't catch it prior to committing.

Comment (jcollard) [reply vagrant/README.md]: Oh, you're suggesting amending the previous commit? Sure.

**Comment by lourinaldi** [comment]
This is something I wish I'd had when I was working to convert the environment over from Gumby. I'm in favor of it.  👍

**Comment by treznick** [comment]
👍
